src/main/java/net/openhft/chronicle/wire/internal/MethodWriterClassNameGenerator.java

```diff
@@ -0,0 +1,50 @@
+package net.openhft.chronicle.wire.internal;
+
+import net.openhft.chronicle.wire.Base32LongConverter;
+import net.openhft.chronicle.wire.LongConverter;
+import net.openhft.chronicle.wire.WireType;
+import net.openhft.hashing.LongHashFunction;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Set;
+
+public class MethodWriterClassNameGenerator {
+
+    private static final LongHashFunction HASH_FUNCTION = LongHashFunction.city_1_1();
+    private static final int MAXIMUM_CLASS_NAME_LENGTH = 255 - ".class".length();
+    /**
+     * This long converter must produce only characters that are safe to appear in filenames
+     */
+    private static final LongConverter CLASSNAME_HASH_LONG_CONVERTER = Base32LongConverter.INSTANCE;
+    private static final int MAX_LENGTH_OF_HASH = CLASSNAME_HASH_LONG_CONVERTER.asText(Long.MIN_VALUE).length();
+
+    @NotNull
+    public String getClassName(@NotNull Set<Class<?>> interfaces, @Nullable String genericEvent, boolean metaData, boolean intercepting, @NotNull WireType wireType) {
+        final StringBuilder sb = new StringBuilder();
+
+        interfaces.forEach(i -> {
+            if (i.getEnclosingClass() != null)
+                sb.append(i.getEnclosingClass().getSimpleName());
+            sb.append(i.getSimpleName());
+        });
+        int endOfInterfacesIndex = sb.length();
+        sb.append(genericEvent == null ? "" : genericEvent);
+        sb.append(metaData ? "MetadataAware" : "");
+        sb.append(intercepting ? "Intercepting" : "");
+        sb.append(toFirstCapCase(wireType.toString().replace("_", "")));
+        sb.append("MethodWriter");
+        if (sb.length() > MAXIMUM_CLASS_NAME_LENGTH) {
+            int firstIndexTruncated = endOfInterfacesIndex - (sb.length() - MAXIMUM_CLASS_NAME_LENGTH) - MAX_LENGTH_OF_HASH;
+            final long hashOfTruncatedInterfaces = HASH_FUNCTION.hashChars(sb, firstIndexTruncated, endOfInterfacesIndex - firstIndexTruncated);
+            sb.delete(firstIndexTruncated, endOfInterfacesIndex);
+            sb.insert(firstIndexTruncated, Base32LongConverter.INSTANCE.asText(hashOfTruncatedInterfaces));
+        }
+        return sb.toString();
+    }
+
+    @NotNull
+    private String toFirstCapCase(@NotNull String name) {
+        return Character.toUpperCase(name.charAt(0)) + name.substring(1).toLowerCase();
+    }
```

Comment:
Extracted out so we can test it properly, also potentially could be configurable if there were ever reason to want different generated names.

Reply:
It's the generated class name you are interested in, that should be testable.

Reply:
It is, that's exactly what MethodWriterClassNameGeneratorTest does. (see... every test case in that test)

Do you mean you want the package testable as well? I don't think that's as important as it is pretty much static.

Reply:
(see [the test](https://github.com/OpenHFT/Chronicle-Wire/pull/644/files#diff-dda75cf859856e30c3a4e3004c0368cb1a23e447196272bfea3e75fdcfda0aa2))